python/ray/data/_internal/execution/streaming_executor.py

```diff
@@ -312,6 +320,18 @@ def _report_current_usage(
         if self._global_info:
             self._global_info.set_description(resources_status)
 
+    def _update_dataset_metrics(self, op_state):
+        metrics = op_state.op.get_metrics()
+        self._stats_actor.inc_bytes_spilled.remote(
```

Comment:
nit, it'd be better to not expose the stats actor, we can hide the implementation details in a util function in `stats.py`.

Reply:
I moved the statsactor calls to a util function, but we still have the streamingexecutor get the statsactor. Unless `_get_or_create_stats_actor` is cheap and can be called every time we update?

Reply:
It looks like these [tests](https://buildkite.com/ray-project/premerge/builds/7918#_) could be failing because of this global stats actor? Is there any way to get around this?